manager/state/raft/transport/peer.go

```diff
@@ -132,6 +138,56 @@ func (p *peer) resolveAddr(ctx context.Context, id uint64) (string, error) {
 	return resp.Addr, nil
 }
 
+// Returns the raft message struct size (not including the payload size) for the given raftpb.Message.
+// The payload is typically the snapshot or append entries.
+func raftMessageStructSize(m *raftpb.Message) int {
+	return (&api.ProcessRaftMessageRequest{Message: m}).Size() - len(m.Snapshot.Data)
+}
+
+// Returns the max allowable payload based on MaxRaftMsgSize and
+// the struct size for the given raftpb.Message and
+func raftMessagePayloadSize(m *raftpb.Message) int {
+	return GrpcMaxMsgSize - raftMessageStructSize(m)
```

Comment:
dumb question: is GrpcMaxMsgSize actually the optimal message size? Or would a smaller message perform better?

Reply:
GrpcMaxMsgSize is probably fine for now.